### PR TITLE
Add basic go build CI workflow

### PR DESCRIPTION
**Pull request by Qinusty**
This will help avoid build failures being missed during PR review

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '1.16.5'
+    - run: go build -o scheduler
+      working-directory: scheduler
+    - run: go build -o worker cmd/worker.go
+      working-directory: analysis
```
Comment (tom--pollard) [inline .github/workflows/build.yml:18]: If you add `./...` to the build command in the analysis directory it should recursively build all the sub directories

Comment (Qinusty) [reply .github/workflows/build.yml:18]: This doesn't work due to the multiple files redeclaring the `main` package under `cmd/`. 

```
# github.com/ossf/package-analysis/analysis/cmd
cmd/fillindexes.go:27:6: main redeclared in this block
        previous declaration at cmd/analyze.go:30:6
cmd/worker.go:88:6: main redeclared in this block
        previous declaration at cmd/fillindexes.go:27:6
```

Comment (tom--pollard) [reply .github/workflows/build.yml:18]: Ah, fair enough!

**Review by oliverchang — APPROVED** [review]
Nice, thanks!

**Pull request merged**